test/jdk/java/awt/ScrollPane/ScrollPaneLimitation/ScrollPaneLimitation.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2000, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4046446
+  @summary Tests 16-bit limitations of scroll pane, child's position and size and mouse coordinates
+  @key headful
+*/
+
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.ScrollPane;
+
+import java.awt.event.InputEvent;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseAdapter;
+
+import java.util.Properties;
+
+public class ScrollPaneLimitation {
+    final static int SCROLL_POS = 50000;
+    public static boolean mouseWasPressed = false;
+    public static Component child = null;
+    private Object lock = new Object();
+    public Frame frame;
+    Robot robot;
+    Point p;
+    ScrollPane pane;
+
+    public static void main(String[] args) throws Exception {
+        ScrollPaneLimitation scrollTest = new ScrollPaneLimitation();
+
+        scrollTest.init();
+        scrollTest.start();
+    }
+
+    public void init() throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            frame = new Frame("Scroll Pane Limitation");
+            frame.setLayout(new BorderLayout());
+        });
+        robot = new Robot();
+    }
+
+    public void start() throws Exception {
+
+        // Check for Win32 environment. This test is only for Win32.
+        // Even more: not for Windows 98: it doesn't support scroll children that long.
+        try {
+            Properties prop = System.getProperties();
+
+            String os = prop.getProperty("os.name", "");
+            if (os != null &&
+                    (os.indexOf("98") == -1) &&
+                    (os.startsWith("Win") || os.startsWith("win") ||
+                            os.startsWith("WIN"))) {
```

Comment:
Should use `OSInfo`?

`@requires` tag should be added to the test so that it's not selected for other platforms.

Reply:
Updated.